LESSONS.md

```diff
@@ -0,0 +1,7 @@
+2020-08-18 08:49:56
+
+# Lessons Learned
+
+-   TypeSet vs TypeList
+-   Structure files vs single file per resource/data source
```

Comment:
Correct on HashiCorp suggesting not to do it, but I haven't found any issues with the pattern. I think I might prefer having the structures files because if we moved those functions to the main resource files, some of those files would be impossible to navigate. I like seeing functionality side-by-side in my editor too and having the two separate files helps me there I think. All that being said, I am totally okay with whichever pattern we want to go with.

Reply:
All said I am a fan of co-locating code in the same files.  This pattern propagated from the Extending Terraform guide at some point early on.